### ISSUE TITLE
web3modal/react 2.0.0 incompatible with wagmi 0.10.13

### ISSUE THREAD

**Issue by paintoshi** [WalletConnect/web3modal#839]
**Describe the bug**
Getting incompatibility error when using web3modal/react 2.0.0 together with wagmi 0.10.13. Specifically in the walletConnectProvider:

```
const { provider } = configureChains(chains, [
  walletConnectProvider({ projectId }),
])
```

```
Type 'ChainProviderFn<JsonRpcProvider, WebSocketProvider, Chain>' is not assignable to type 'ChainProviderFn<JsonRpcProvider, WebSocketProvider, { readonly id: 42161; readonly name: "Arbitrum One"; readonly network: "arbitrum"; readonly nativeCurrency: { ...; }; readonly rpcUrls: { ...; }; readonly blockExplorers: { ...; }; readonly contracts: { ...; }; } | { ...; } | { ...; } | { ...; }>'.
  Type '{ chain: Chain; provider: () => ProviderWithFallbackConfig<JsonRpcProvider>; webSocketProvider?: (() => WebSocketProvider) | undefined; } | null' is not assignable to type '{ chain: { readonly id: 42161; readonly name: "Arbitrum One"; readonly network: "arbitrum"; readonly nativeCurrency: { readonly name: "Ether"; readonly symbol: "ETH"; readonly decimals: 18; }; readonly rpcUrls: { ...; }; readonly blockExplorers: { ...; }; readonly contracts: { ...; }; } | { ...; } | { ...; } | { ......'.
    Type '{ chain: Chain; provider: () => ProviderWithFallbackConfig<JsonRpcProvider>; webSocketProvider?: (() => WebSocketProvider) | undefined; }' is not assignable to type '{ chain: { readonly id: 42161; readonly name: "Arbitrum One"; readonly network: "arbitrum"; readonly nativeCurrency: { readonly name: "Ether"; readonly symbol: "ETH"; readonly decimals: 18; }; readonly rpcUrls: { ...; }; readonly blockExplorers: { ...; }; readonly contracts: { ...; }; } | { ...; } | { ...; } | { ......'.
      Types of property 'chain' are incompatible.
        Type 'Chain' is not assignable to type '{ readonly id: 42161; readonly name: "Arbitrum One"; readonly network: "arbitrum"; readonly nativeCurrency: { readonly name: "Ether"; readonly symbol: "ETH"; readonly decimals: 18; }; readonly rpcUrls: { ...; }; readonly blockExplorers: { ...; }; readonly contracts: { ...; }; } | { ...; } | { ...; } | { ...; }'
```

**Workaround**
Hence it won't work following the installation instructions on https://docs.walletconnect.com/2.0/web3modal/react/installation, but must specifically use "yarn add wagmi@0.10.12"

**Comment by 0xAsimetriq** [comment]
Ty for report, just upgraded our examples and was able to re-create the issue. But since we are returning default jsonRpcProvider from `walletConnectProvider` this looks like a type bug introduced in latest version of wagmi. I'm also getting same one when using their `publicProvider`, so in theory you can just `// @ts-expect-error` comment it for now.

I'll close the issue here, as it is an upstream dependency fix in wagmi for types. I've also pinged their team to have a look 👍  

<img width="1084" alt="Screenshot 2023-01-23 at 10 14 47" src="https://user-images.githubusercontent.com/3154053/213993104-38277c2f-f7ce-4fb2-bce2-cda3e68a6ed3.png">

**Comment by 0xAsimetriq** [comment]
@paintoshi should be fixed in wagmi `0.10.14` 👍
